### PR TITLE
feat: Adding market flags proposal spec

### LINKED ISSUE #2101: Add specification for governance proposal to add flags to a market
Create a spec to add flags/tags to a market via a governance vote.

- Flags should be added/removed with a proposal which exclusively adds/removes certain tags without affecting others on the market
- They should not be changeable with a normal market amendment
- ELS on a market does not allow one to vote on them with additional weight

### PR DESCRIPTION
**Pull request by TomMcL**
Closes #2101

### CODE REVIEW

File: protocol/0028-GOVE-governance.md [hunk 1]
```diff
@@ -249,6 +249,19 @@ The following are immutable and cannot be changed:
 - position decimal places
 - `settlementAsset`
 
+### 2.1 Change market flags
+
+Separately to a general market change proposal, proposers can create a proposal to add or remove lists of free-text string flags to a market. Although these proposals change only a single market, it is possible that these tags will be used to control external behaviour of systems interacting with the vega chain, therefore they do not allow for voting using ELS and instead must be voted through by the general pool of token holders as with any other proposal. The maximum length of a flag is specified by the network parameter `governance.proposal.market.maxFlagLength`. The flags on a market are controlled by a single proposal type, `UpdateFlags` with a structure akin to:
+
+```proto
+message UpdateFlags {
```
Comment (davidsiska-vega) [inline protocol/0028-GOVE-governance.md]: should there be marketId field here?

File: protocol/0028-GOVE-governance.md [hunk 2]
```diff
@@ -249,6 +249,19 @@ The following are immutable and cannot be changed:
 - position decimal places
 - `settlementAsset`
 
+### 2.1 Change market flags
+
+Separately to a general market change proposal, proposers can create a proposal to add or remove lists of free-text string flags to a market. Although these proposals change only a single market, it is possible that these tags will be used to control external behaviour of systems interacting with the vega chain, therefore they do not allow for voting using ELS and instead must be voted through by the general pool of token holders as with any other proposal. The maximum length of a flag is specified by the network parameter `governance.proposal.market.maxFlagLength`. The flags on a market are controlled by a single proposal type, `UpdateFlags` with a structure akin to:
```
Comment (davidsiska-vega) [inline protocol/0028-GOVE-governance.md]: I think we'd like the voting on these to be a separate category with lengths and participations so:
```
governance.proposal.updateMarketFlags.maxClose
720h
governance.proposal.updateMarketFlags.maxEnact
720h
governance.proposal.updateMarketFlags.minClose
168h
governance.proposal.updateMarketFlags.minEnact
168h
governance.proposal.updateMarketFlags.minProposerBalance
20,000.00
governance.proposal.updateMarketFlags.minVoterBalance
1.00
governance.proposal.updateMarketFlags.requiredMajority
66%
governance.proposal.updateMarketFlags.requiredParticipation
```

**Review by davidsiska-vega — COMMENTED** [review]
It may be good to add a mention of this into batch proposals too i.e. this can be included within a batch.

File: protocol/0001-MKTF-market_framework.md [hunk 3]
```diff
@@ -80,6 +80,7 @@ Data:
 - **Name:** full and fairly descriptive name for the instrument.
 - **Metadata fields:** A series of arbitrary strings that can be used in clients
 - **Product:** a reference to or instance of a fully specified product, including all required product parameters for that product.
+- **Flags:** A list of string flags assigning the market to certain categories. Each of these is a free text field of a network parameter defined maximum length. These are managed through a separate proposal type and cannot be set at market creation or changed in a normal market update proposal.
```
Comment (davidsiska-vega) [inline protocol/0001-MKTF-market_framework.md]: Tags?

**Review by davidsiska-vega — COMMENTED** [review]
One flags -> tags seems to have escaped.

File: protocol/0001-MKTF-market_framework.md (hunk 3 above)
Comment (TomMcL) [reply protocol/0001-MKTF-market_framework.md]: It has been culled

**Review by davidsiska-vega — DISMISSED** [review]

**Review by davidsiska-vega — APPROVED** [review]

**Pull request merged**